source/geod24/LocalRest.d

```diff
@@ -553,31 +711,78 @@ public final class RemoteAPI (API, alias S = VibeJSONSerializer!()) : API
             // Control the node behavior
             Control control;
 
+            struct AwaitingMessages
+            {
+                /// Message
+                Variant var;
+                /// Originating `Connection`
+                Connection conn;
+            }
             // we need to keep track of messages which were ignored when
             // node.sleep() was used, and then handle each message in sequence.
-            Variant[] await_msgs;
+            AwaitingMessages[] await_msgs;
 
             scheduler.start(() {
-                while (1)
+                while (true)
                 {
+                    Connection new_conn;
+                    Response res;
                     Variant msg;
-                    if (chn.read(msg, 10.msecs))
+
+                    C.SelectEntry[] read_list;
+                    C.SelectEntry[] write_list;
```

Comment:
This will be quite GC-heavy with all the allocations. How about moving it oustide the while loop, resetting the lengths to zero and using `assumeSafeAppend`?